framework/include/meshgenerators/BreakMeshByBlockGenerator.h

```diff
@@ -41,20 +41,29 @@ class BreakMeshByBlockGenerator : public BreakMeshByBlockGeneratorBase
   /// where not necessary.
   subdomain_id_type blockRestrictedElementSubdomainID(const Elem * elem);
 
+  /// Return true if block_one and block_two are found in users' provided block_pairs list
+  bool findBlockPairs(subdomain_id_type block_one, subdomain_id_type block_two);
+
   /// the mesh to modify
   std::unique_ptr<MeshBase> & _input;
-  /// the blocks to split the mesh on
-  const std::vector<SubdomainID> _block;
+  // vector of subdomains surrounding which interfaces will be generated
+  const std::vector<SubdomainID> _surrounding_blocks;
+  /// vector of subdomain pairs between which interfaces will be generated.
+  const std::vector<std::vector<SubdomainID>> _block_pairs;
```

Comment:
If you used a std::set<std::pair> instead of std::vector<std::vector> it would prevent duplicate SubdomainID pairs

Reply:
good point. would `std::pair` work as a parameter type?

Reply:
we should also make _surrounding_blocks a std::set<SubdomainID>

Reply:
I'm not sure.  It seems like addParam is always taken in as a string and converted based on the type.  However I can't find any example of it within MOOSE or BISON.

Reply:
> I'm not sure. It seems like addParam is always taken in as a string and converted based on the type. However I can't find any example of it within MOOSE or BISON.

I tried to switch to `set`, but I got some compiling errors. 

@dschwen do you know if we can use `set` and/or `pair` as `Parameter` type?